app/controllers/db_files_controller.rb

```diff
@@ -0,0 +1,12 @@
+class DbFilesController < ApplicationController
+  skip_before_action :check_admin
+
+  def show
+    # TODO: Generate an external key instead of using internal id
+    # because we have no access control on a file
```

Comment:
Hmm, this is a pretty expensive operation on the db, but I don't think this matters?
```
SELECT count(*) > 1 FROM submissions JOIN db_files ON submissions.db_file_id = db_files.id WHERE submissions.user_id = current_user.id
```